model/product/product-controller.js

```diff
@@ -0,0 +1,110 @@
+const Controller = require('../../lib/controller');
+const productModel  = require('./product-facade');
+const util       = require('util');
+const _          = require('underscore')._;
+
+const findSchema = {
+  ean: {
+    in: 'query',
+    optional: true,
+    isEan: {
+      errorMessage: '%0 is not a valid EAN'
+    }
+  },
+  name:{
+    in: 'query',
+    optional: true
+  }
+};
+
+class ProductController extends Controller {
```

Comment:
Avoid code duplication, refactor the param check to prevent rewriting it in every method.